test/ContractInteractor.test.ts

```diff
@@ -0,0 +1,164 @@
+const DEFAULT_CHAIN_ID = 33;
+
+describe('ContractInteractor', () => {
+    const defaultConfig: EnvelopingConfig = {
+        preferredRelays: [],
+        onlyPreferredRelays: false,
+        relayLookupWindowParts: 1,
+        relayLookupWindowBlocks: DEFAULT_LOOKUP_WINDOW_BLOCKS,
+        gasPriceFactorPercent: GAS_PRICE_PERCENT,
+        minGasPrice: 60000000, // 0.06 GWei
+        maxRelayNonceGap: MAX_RELAY_NONCE_GAP,
+        sliceSize: 3,
+        relayTimeoutGrace: DEFAULT_RELAY_TIMEOUT_GRACE_SEC,
+        methodSuffix: '',
+        jsonStringifyRequest: false,
+        chainId: DEFAULT_CHAIN_ID,
+        relayHubAddress: constants.ZERO_ADDRESS,
+        deployVerifierAddress: constants.ZERO_ADDRESS,
+        relayVerifierAddress: constants.ZERO_ADDRESS,
+        forwarderAddress: constants.ZERO_ADDRESS,
+        smartWalletFactoryAddress: constants.ZERO_ADDRESS,
+        logLevel: 0,
+        clientId: '1'
+    };
+    let mockWeb3Provider: Web3Provider;
+    let contractInteractor: ContractInteractor;
+
+    before(() => {
+        mockWeb3Provider = stubInterface<Web3Provider>();
+        contractInteractor = new ContractInteractor(
+            mockWeb3Provider,
+            defaultConfig
+        );
+    });
+
+    describe('verifyForwarder', () => {
+        let _createForwarderStub: sinon.SinonStub;
+        const fakeIForwarderInstance: sinon.SinonStubbedInstance<IForwarderInstance> &
+            IForwarderInstance = stubInterface<IForwarderInstance>();
+        const fakeSuffixData = 'fakeSuffix';
+        const fakeRelayRequest: RelayRequest = {
+            request: {
+                to: 'fake_address',
+                data: 'fake_data',
+                gas: '1'
+            } as ForwardRequest,
+            relayData: {
+                gasPrice: '0',
+                callForwarder: 'fake_address'
+            } as RelayData
+        };
+        const fakeSignature = 'fake_signature';
+
+        before(() => {
+            _createForwarderStub = sinon
+                .stub(contractInteractor, '_createForwarder')
+                .callsFake(() => Promise.resolve(fakeIForwarderInstance));
+        });
+
+        it('should use _createForwarder once', async () => {
+            await expect(
+                contractInteractor.verifyForwarder(
+                    fakeSuffixData,
+                    fakeRelayRequest,
+                    fakeSignature
+                )
+            ).to.eventually.be.undefined;
+            expect(contractInteractor._createForwarder).to.have.been.called;
+        });
+
+        it('should use forwarder.verify once', async () => {
+            await expect(
+                contractInteractor.verifyForwarder(
+                    fakeSuffixData,
+                    fakeRelayRequest,
+                    fakeSignature
+                )
+            ).to.eventually.be.undefined;
+            expect(fakeIForwarderInstance.verify).to.have.been.called;
```

Comment:
Here we're not checking how many time the function has been called according with the test description